editor/plugins/org.fusesource.ide.project/src/org/fusesource/ide/project/camel/ui/CamelFacetInstallationPage.java

```diff
@@ -0,0 +1,191 @@
+				}
+			});
+		}
+	}
+
+	public void createControl(Composite parent) {
+		setControl(createTopLevelComposite(parent));
+	}
+	
+	private void changePageStatus(){
+		boolean webfound = hasWebFacet();
+		boolean shouldUpdateStructure = model.getBooleanProperty(UPDATE_PROJECT_STRUCTURE);
+		boolean displayContentFolder = webfound || shouldUpdateStructure;
+
+		if(displayContentFolder && !validFolderName(contentFolder.getText())){
+			setErrorMessage(Messages.NewCamelProject_FacetInstallationPage_ContentRootError);
+		} else {
+			setErrorMessage(null);
+		}
+		setPageComplete(isPageComplete());
+	}
+	
+	private boolean validFolderName(String folderName) {
+		IWorkspace ws = ResourcesPlugin.getWorkspace();
+		return ws.validateName(folderName, IResource.FOLDER).isOK();
+	}
+
+	
+	protected Composite createTopLevelComposite(Composite parent) {
+		final Composite composite = new Composite(parent, SWT.NONE);
+		composite.setLayout(new GridLayout(1, false));
+		createProjectGroup(composite);
+//		//set page status
+		changePageStatus();
+		
+		return composite;
+	}
+
+	private void createProjectGroup(Composite parent){
+		Composite prjGroup = new Composite(parent, SWT.NONE);
+		prjGroup.setLayout(new GridLayout(1, false));
+		prjGroup.setLayoutData(new GridData(GridData.FILL_HORIZONTAL));
+		
+		this.contentRootLabel = new Label(prjGroup, SWT.NONE);
+		this.contentRootLabel.setText(Messages.NewCamelProject_ContentRootLabel);
+		this.contentRootLabel.setLayoutData(new GridData(GridData.FILL_HORIZONTAL));
+
+		this.contentFolder = new Text(prjGroup, SWT.BORDER);
+		this.contentFolder.setLayoutData(new GridData(GridData.FILL_HORIZONTAL));
+		this.contentFolder.setData("label", this.contentRootLabel); //$NON-NLS-1$ // wtf??
+		this.contentFolder.setText(model.getStringProperty(CAMEL_CONTENT_FOLDER));
+		contentFolder.addModifyListener(new ModifyListener(){
+			public void modifyText(ModifyEvent e){
+				 String content = contentFolder.getText();
+				 if(content != null && !content.equals("")){ //$NON-NLS-1$
+					 model.setProperty(CAMEL_CONTENT_FOLDER, content);
+				 }
+				 changePageStatus();
+			}
+		});
+
+		this.dslLabel = new Label(prjGroup, SWT.NONE);
+		this.dslLabel.setText(Messages.NewCamelProject_DSLLabel);
+		this.dslLabel.setLayoutData(new GridData(GridData.FILL_HORIZONTAL));
+
+		this.dslSelectionCombo = new Combo(prjGroup, SWT.DROP_DOWN | SWT.READ_ONLY);
+		this.dslSelectionCombo.setLayoutData(new GridData(GridData.FILL_HORIZONTAL));
+		this.dslSelectionCombo.setItems(new String[] {"Blueprint", "Spring", "Routes"});
+		dslSelectionCombo.addSelectionListener( new SelectionAdapter() {
+			public void widgetSelected(SelectionEvent e) {
+				String dsl = dslSelectionCombo.getItem(dslSelectionCombo.getSelectionIndex());
+				model.setProperty(CAMEL_DSL, dsl);
+				changePageStatus();
+			}
+		});
+		this.dslSelectionCombo.select(0);
+		
+		updateWidgetsFromModel();
+	}
+	
+	
+	private boolean hasWebFacet() {
+		boolean webfound = false;
```

Comment:
this code is duplicated in 3 places. 2 times with the exact same jst.web and one time for another id. We can factorize it